### PR TITLE
feature(cli): delete command for models and bentos

### PR DESCRIPTION
**Pull request by larme**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by parano** [comment]
could you take a look into codestyle_check CI? it's still failing

File: bentoml/_internal/cli/bento_management.py [hunk 1]
```diff
@@ -148,7 +148,22 @@ def delete(
...
-        pass
+
+        def delete_target(target):
+            to_delete_bentos = bento_store.list(target)
+
+            for bento in to_delete_bentos:
+                if yes:
+                    if_delete = True
+                else:
+                    if_delete = click.confirm(f"delete bento {bento.tag}?")
```
Comment (parano) [inline bentoml/_internal/cli/bento_management.py:159]:
```suggestion
                    if_delete = click.confirm(f'Delete bento "{bento.tag}"?')
```

File: bentoml/_internal/cli/bento_management.py [hunk 2]
```diff
@@ -148,7 +148,22 @@ def delete(
...
         * Bulk delete multiple bento bundles by name and version, separated by ",", e.g.: `benotml delete Irisclassifier:v1,MyPredictService:v2`
         """  # noqa
-        pass
+
+        def delete_target(target):
+            to_delete_bentos = bento_store.list(target)
+
+            for bento in to_delete_bentos:
+                if yes:
+                    if_delete = True
```
Comment (parano) [inline bentoml/_internal/cli/bento_management.py:157]: rename `if_delete` to `deletion_confirmed`?

File: bentoml/_internal/cli/bento_management.py [hunk 3]
```diff
@@ -148,7 +148,22 @@ def delete(
...
+                else:
+                    if_delete = click.confirm(f"delete bento {bento.tag}?")
+
+                if if_delete:
+                    click.echo(f"deleting bento {bento.tag}")
+                    bento_store.delete(bento.tag)
```
Comment (parano) [inline bentoml/_internal/cli/bento_management.py:163]:
```suggestion
                    bento_store.delete(bento.tag)
                    click.echo(f"{bento} deleted")
```

File: bentoml/_internal/cli/model_management.py [hunk 4]
```diff
@@ -150,7 +150,22 @@ def delete(
...
         * Bulk delete multiple models by name and version, separated by ",", e.g.: `benotml model delete Irisclassifier:v1,MyPredictService:v2`
         """  # noqa
-        pass
+
+        def delete_target(target):
+            to_delete_models = model_store.list(target)
+
+            for model in to_delete_models:
+                if yes:
+                    if_delete = True
```
Comment (parano) [inline bentoml/_internal/cli/model_management.py:159]: ditto

File: bentoml/_internal/cli/model_management.py [hunk 5]
```diff
@@ -150,7 +150,22 @@ def delete(
...
+                else:
+                    if_delete = click.confirm(f"delete bento {model.tag}?")
+
+                if if_delete:
+                    click.echo(f"deleting bento {model.tag}")
```
Comment (parano) [inline bentoml/_internal/cli/model_management.py:164]: typo: ~bento~model

Comment (parano) [inline bentoml/_internal/cli/model_management.py:164]: similarly to above, I'd suggest remove the `deleting` line, and add a `deleted` message after the deletion call.

**Comment by larme** [comment]
> could you take a look into codestyle_check CI? it's still failing

Currently codestyle check failing because we have unimplemented commands, should be ok after we finished them

Other change suggestions are done

File: bentoml/_internal/cli/model_management.py [hunk 6]
```diff
@@ -150,7 +150,22 @@ def delete(
...
+
+            for model in to_delete_models:
+                if yes:
+                    delete_confirmed = True
+                else:
+                    delete_confirmed = click.confirm(f"delete model {model.tag}?")
+
+                if delete_confirmed:
+                    model_store.delete(model.tag)
+                    click.echo(f"{model.tag} deleted")
```
Comment (parano) [inline bentoml/_internal/cli/model_management.py:165]:
```suggestion
                    click.echo(f"{model} deleted")
```

File: bentoml/_internal/cli/bento_management.py [hunk 7]
```diff
@@ -148,7 +148,22 @@ def delete(
...
+
+            for bento in to_delete_bentos:
+                if yes:
+                    delete_confirmed = True
+                else:
+                    delete_confirmed = click.confirm(f"delete bento {bento.tag}?")
+
+                if delete_confirmed:
+                    bento_store.delete(bento.tag)
+                    click.echo(f"{bento.tag} deleted")
```
Comment (parano) [inline bentoml/_internal/cli/bento_management.py:163]:
```suggestion
                    click.echo(f"{bento} deleted")
```

**Pull request merged**